app/classifier/services.py

```diff
@@ -53,13 +54,18 @@ def train(self, dataloaders, criterion, optimizer, scheduler):
             if phase == 'training':
               loss.backward()
               optimizer.step()
-          # statistics
+          # statistics  
           running_loss += loss.item() * inputs[0].size(0)
           running_corrects += torch.sum(preds == labels.data)
         if phase == 'training' and scheduler is not None:
           scheduler.step()
         epoch_loss = running_loss / len(dataloaders[phase].dataset)
-        epoch_acc = running_corrects.double() / len(dataloaders[phase].dataset)
+        epoch_acc = f1_score(labels.tolist(), preds.tolist(), average="macro")
```

Comment:
Big mistake here, only taking into account those labels and predictios corresponding to last batch. Must compare whole epochs, else the measure will be biased.

Reply:
Solved in https://github.com/sperezacuna/oracle-challenge-f2/pull/9/commits/fefc385cca7ca1a7fa603478cced16f2d138485f